### ISSUE TITLE
Input shaping with adxl345 

### ISSUE THREAD

**Issue by letiguja** [gaifeng8864/klipper-on-android#23]
Can I use adxl345 to do input shaping and if it is possible how would i do it.

**Comment by gaifeng8864** [comment]
> Can I use adxl345 to do input shaping and if it is possible how would i do it.

Yes, you can use ADXL345. However, you need to connect it to the hardware SPI interface on the printer motherboard through the SPI interface (software SPI usually has insufficient communication speed or occupies too much CPU of the printer control board), and then add the following configuration information to the configuration file printer.cfg.

[adxl345]
cs_pin: PA4
spi_bus: spi1
axes_map: -x,-y,z

[resonance_tester]
accel_chip: adxl345
probe_points:
135,135,10 # an example

Note:
cs_pin: PA4
spi_bus: spi1
You need to determine it according to the interface definition of the printer motherboard. For details, please refer to the official pin definition document of the printer control board.

For other options, please refer to the instructions in the official document of klipper and modify them according to your own control board.
